subsys/mpsl/cx/Kconfig

```diff
@@ -80,10 +81,34 @@ config MPSL_CX_BT_1WIRE
 	  Radio Coexistence interface implementation using a simple 1-wire PTA
 	  implementation for co-located radios.
 
+config MPSL_CX_SOFTWARE
+	bool "Software Coexistence"
+	help
+	  Radio Coexistence interface implementation using software APIs to
+	  configure currently granted radio operations.
+
 endchoice # MPSL_CX_CHOICE
 
 module=MPSL_CX
 module-str=MPSL_CX
 source "${ZEPHYR_BASE}/subsys/logging/Kconfig.template.log_config"
 
 endif	# MPSL_CX
+
+config MPSL_CX_SOFTWARE_RPC
+	bool
+
+config MPSL_CX_SOFTWARE_RPC_CLIENT
```

Comment:
Since `MPSL_CX_SOFTWARE` is one of the options in `MPSL_CX_CHOICE` along with hardware PTA implementations, I'd suggest creating a Kconfig interface similar to the existing approach. For multicore architectures, that means setting the same options for the application core and the core that controls the RADIO peripheral, e.g.
```
west build -p -b nrf5340dk/nrf5340/cpuapp -- -DCONFIG_MPSL_CX=y -DCONFIG_MPSL_CX_NRF700X=y
-Dhci_ipc_CONFIG_MPSL_CX=y -Dhci_ipc_CONFIG_MPSL_CX_NRF700X=y
```
To align with this approach I'd suggest to make `MPSL_CX_SOFTWARE_RPC_...` symbols hidden. Their values would need to be calculated based on the target SoC similarly to `MPSL_CX_PIN_FORWARDER`. Then, the application would only set `CONFIG_MPSL_CX=y` and `CONFIG_MPSL_CX_SOFTWARE=y` and not care about the target platform.

Note that the file you've introduced: `applications/protocols_serialization/client/snippets/coex/coex.conf` needs to set the RPC option explicitly, which breaks the independency from target platform. The snippet by design cannot be applied to single core architectures, which kind of defeats its purpose.

Reply:
To give you more context, we're building a prototype in which application runs on a different SoC than the OpenThread stack, and they communicate with each other using UART (and nRF RPC). That's why I cannot rely on automatic detection of the client and server role because a user may select any SoC for either role.

Well, I could probably infer that if `CONFIG_MPSL_CX_SOFTWARE && !MPSL` then it means `MPSL_CX_SOFTWARE_RPC_CLIENT` but not sure if this isn't more cryptic. Let me know if you have a better idea having the context :)

Reply:
Thanks for the additional context. I still think that if `MPSL_CX_SOFTWARE` is one of the options in `MPSL_CX_CHOICE`, the interface should be the same for consistency. `CONFIG_MPSL_CX_SOFTWARE && !CONFIG_MPSL` doesn't seem cryptic at all to me, quite the opposite actually ;) it's a good proposal.

Reply:
Implemented the described pattern (I hope).